docs/key-issues/gpus.md

```diff
@@ -180,3 +204,85 @@ Potential solutions on the PyPI side include:
 - make an environment marker or selector package approach work,
 - improve interoperability with other package managers, in order to be able to
   declare a dependency on a CUDA or ROCm version as externally provided,
+
+
+## Additional Notes on CUDA Compatibility
+
+Compatibility across CUDA versions is a common problem with numerous pitfalls to be aware of.
+There are three primary components of CUDA:
+
+1. The CUDA Toolkit (CTK): This component includes the CUDA runtime library
+   (libcudart.so) along with a range of other libraries and tools including
+   math libraries like cuBlas. libcudart.so and a few other core headers and
+   libraries compose the bare minimum required to compile CUDA code.
+2. The user-mode driver: This is the libcuda.so library. This library is
+   required to actually run CUDA code.
+3. The kernel-mode driver: The nvidia.ko file. This constitutes what is
+   typically considered a "driver" in common parlance when referring to other
+   peripherals connected to a computer.
+
+CUDA developers and users generally do not need to be aware of the kernel-mode driver.
+For the rest of this section, when referring to the "driver" we will always be referring to the user-mode driver libcuda.so.
+
+The CUDA runtime library makes no forward or backwards compatibility
+guarantees, meaning that libraries that dynamically link to the CUDA runtime
+may not work correctly if they are run on a system with a different CUDA
+runtime shared library than the one they were compiled against. In this scenario,
+users are responsible for having the right CUDA runtime library installed.  As
+a result, the official CUDA recommendation is to statically link the CUDA
+runtime (see
+[here](https://docs.nvidia.com/cuda/cuda-c-best-practices-guide/index.html#recommendations-for-building-a-minor-version-compatible-library)
+and
+[here](https://docs.nvidia.com/cuda/cuda-c-best-practices-guide/index.html#distributing-the-cuda-runtime-and-libraries)
+for more information).
+
+CUDA drivers have always been backwards compatible. Any code that runs when
+some driver version X is installed will always work correctly when run with
+some newer driver version Y>X. As briefly discussed above, as of CUDA 11.0
+CUDA also promises [minor version
+compatibility](https://docs.nvidia.com/deploy/cuda-compatibility/index.html#minor-version-compatibility)
+(MVC). This compatibility guarantees that CUDA code compiled using a certain
+version of the CUDA toolkit will be compatible with any driver version within
+the same major release. This behavior is useful because it is often easier for
```

Comment:
What is "the minimal requirement" in this context?

Reply:
Should this then say something like:

```suggestion
version of the CUDA toolkit will be compatible with any driver version above
the minimal requirement specified for MVC with that toolkit. This behavior is useful because it is often easier for
```

I'm not super happy with my suggestion - it's a bit of a mouthful - but otherwise I think it's a bit confusing what the minimal requirement is or where to look for it.

Reply:
Alternatively we could just link to [the official table](https://docs.nvidia.com/deploy/cuda-compatibility/index.html#default-to-minor-version). Then we don't have to maintain much.

What @leofang said above is indeed a more precise statement, but only because MVC was only imperfectly included in the first few releases of CUDA 11. For future versions I would expect all the above statements to be equivalent because the earliest official release for major version X will have a UMD compatible with all later minor CTKs.